File: framework/src/main/java/com/software/dm/swallow/stormy/algoac/impl/CharacterState.java
package com.software.dm.swallow.stormy.algoac.impl;

import com.software.dm.swallow.stormy.algoac.inter.AbstractState;

/**
 * @author DearM
 * @version v1.0.0.1
 * @Description
 * @date 2016
 */
public class CharacterState extends AbstractState<Character> {

    public CharacterState() {
        super();
    }

    public CharacterState(int depth) {
        super(depth);
    }


    public AbstractState<Character> addState(Character c) {
        AbstractState<Character> abstractState = this.nextStateIgnoreRootState(c);
        if (abstractState == null) {
            abstractState = new CharacterState(this.depth + 1);
            this.success.put(c, abstractState);
        }
        return abstractState;
    }


}

File: framework/src/main/scala/com/software/dm/swallow/stormy/scala/algoac/impl/CharacterState.scala
package com.software.dm.swallow.stormy.scala
package algoac
package impl

import inter.AbstractState


/**
  *
  * Description
  * author DearM
  * date 2016
  *
  * @version v1.0.0.1
  *
  */
class CharacterState(depth: Int) extends AbstractState[Character](depth: Int) {


  override def addState(c: Character): AbstractState[Character] = {
    var abstractState = this.nextStateIgnoreRootState(c)
    if (abstractState == null) {
      abstractState = new CharacterState(this.depth + 1)
      this.success.put(c, abstractState)
    }
    abstractState
  }
}

object CharacterState {

}